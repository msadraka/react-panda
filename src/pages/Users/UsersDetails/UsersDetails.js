/* @flow */

import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { Avatar, Button } from 'antd';
import { get } from 'lodash';

import { load as loadUser } from 'actions/users.action';
import StyleWrapper from './usersDetails.style';

type Props = {
  history: Object,
  params: Object,
  users: Object,
  loadUser: Function
};

type State = {};

// Export this for unit testing more easily
export class UsersDetails extends PureComponent<Props, State> {
  componentDidMount() {
    const { params, loadUser: loadPromise } = this.props;
    console.log('params', params);
    loadPromise(params.id);
  }

  // Back to users list page handler
  handleBack = () => {
    const { history } = this.props;
    history.push('/');
  };

  render() {
    const { users } = this.props;
    const usersDate = get(users, 'data.data', []);

    return (
      <StyleWrapper>
        <Helmet title="New User" />
        <div className="back-to-list">
          <Button
            size="large"
            type="primary"
            shape="round"
            icon="left"
            onClick={this.handleBack}
          >
            Back to list
          </Button>
        </div>

        <div className="user--details">
          <div className="details--avatar">
            <Avatar size={120} src={usersDate.avatar} />
          </div>

          <div className="details--fullname">
            {usersDate.first_name} {usersDate.last_name}
          </div>

          <div className="details--email">{usersDate.email}</div>
        </div>
      </StyleWrapper>
    );
  }
}

const mapStateToProps = ({ users }: Object) => ({ users: users.detail });

const mapDispatchToProps = { loadUser };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UsersDetails);
