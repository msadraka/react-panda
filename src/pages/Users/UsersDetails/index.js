/* @flow */

import React from 'react';
import loadable from '@loadable/component';
import { Loading, ErrorBoundary } from 'components';

const UsersDetails = loadable(() => import('./UsersDetails'), {
  fallback: <Loading />
});

export default (props: { props: Object }) => (
  <ErrorBoundary>
    <UsersDetails {...props} />
  </ErrorBoundary>
);
