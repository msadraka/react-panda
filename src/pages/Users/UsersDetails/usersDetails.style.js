import styled from 'styled-components';

export default styled.div`
  .user--details {
    margin-top: 50px;

    .details--fullname {
      font-size: 22px;
      font-weight: bold;
      margin: 10px 0;
      color: #000;
    }

    .details--email {
      color: #000;
    }
  }

  .back-to-list {
    position: absolute;
    top: 30px;
    left: 30px;
  }
`;
