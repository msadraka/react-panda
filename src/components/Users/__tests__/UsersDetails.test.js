import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import UsersDetails from '../Widgets/UsersDetails';

describe('<UsersDetails />', () => {
  it('renders', () => {
    const mockData = {
      id: 1,
      email: 'george.bluth@reqres.in',
      first_name: 'George',
      last_name: 'Bluth',
      avatar:
        'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg'
    };

    const tree = renderer
      .create(
        <MemoryRouter>
          <UsersDetails user={mockData} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
