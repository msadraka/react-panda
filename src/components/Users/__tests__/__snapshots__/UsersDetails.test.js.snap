// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersDetails /> renders 1`] = `
<div
  className="usersDetailsstyle___default-sc-6wapd6-0 cICFtm"
>
  <div
    className="user-avatar"
  >
    <span
      className="ant-avatar ant-avatar-circle ant-avatar-image"
      style={
        Object {
          "fontSize": 18,
          "height": 120,
          "lineHeight": "120px",
          "width": 120,
        }
      }
    >
      <img
        onError={[Function]}
        src="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
      />
    </span>
  </div>
  <div
    className="user-fullname"
  >
    George
     
    Bluth
  </div>
  <div
    className="user-email"
  >
    george.bluth@reqres.in
  </div>
</div>
`;
